src/Main.enso

```diff
@@ -1,105 +1,193 @@
+
+    my_bool = 1 == 2
+    my_bool = 2 > 0.3
+    my_bool = "foo" == "bar"
+
+    # Vectors:
+    # Create with []
+    my_vec = [1, 2, 3]
+    # Vectors can contain multiple types:
+    my_vec_2 = [1, "Foo", False]
+    # `characters` is a method that creates a vector of characters in text.
+    text_sample.characters 
 
-draw_circle radius=1.0 color='red' x_position=0 y_position=0 =
-    ...
+    # Maps
+    # Key:value data structure.
 
+    empty_map = Map.empty
 
-## Type definitions
+    inserted = empty_map.insert 4 "foo" 
+    item = inserted.get 4 # returns "foo"
+    item_2 = inserted.get 10 # returns Dataflow Error 
+
+## Type definition
+# Types are defined with the `type` keyword,
+# followed by a type name and fields.
 type Geo_Point latitude longitude
 
-## Type definitions
+# Types can also be defined with default values for fields.
 type Geo_Point latitude longitude shape='star'
 
-## Defines a method on an Atom (a defined type)
+# Methods can be defined as Extension Methods, on an existing type (an Atom).
 Geo_Point.convert =
-    result = Geo_Point (this.latitude / 3600000) (this.longitude / 3600000) # `this` refers to the `Geo_Point` object
-    here.my_method result.latitude result.longitude # `here` still refers to the current module
+    # `this` refers to the `Geo_Point` object
+    result = Geo_Point (this.latitude / 3600000) (this.longitude / 3600000)
+    # `here` still refers to the current module
+    here.my_method result.latitude result.longitude
     result
 
 test_types =
-    # Create a Geo_Point
+    # Create a Geo_Point:
     my_point = Geo_Point 52.05 15.2
-    # getters!
+    # Fields can be used as getters:
     my_latitude = my_point.latitude
     my_longitude = my_point.longitude
-    # Pattern matching:
+    # Pattern matching on fields:
     my_number = case my_point of
         Geo_Point ltd lng -> ltd * lng
     # Calling a method on a value
     my_point.convert
 
-# Polyglot Java
+# Types can be nested and combined with pattern matching to provide
+# Option-like types.
+type One_Or_Another
+    type One value
+    type Another value_x value_y
+
+    my_method v = case this of
+        One False -> v || value
+        One True -> v && value
+        Another vx vy -> v * vx * vy
+
+## Polyglot: language interoperation
+
+# Java interoperability
 polyglot java import com.example.MyClass
 
 poly_test =
     # Call a static method
-    result = MyClass.staticMethod [arg1, arg2] # Today
-    result = MyClass.staticMethod arg1 arg2 # A week from now
+    result = MyClass.staticMethod arg1 arg2
     # Create an instance! (call a constructor)
-    my_inst = MyClass.new [arg1, arg2].to_array # Today
-    my_inst = MyClass.new arg1 arg2 # A week from now
+    my_inst = MyClass.new arg1 arg2 
     # Call an instance method
-    my_inst.myMethod [1, 2, 3] # Today
-    my_inst.myMethod 1 2 3 # Future
+    my_inst.myMethod 1 2 3 
 
 # Polyglot (R | Python | Ruby | JS | Whatever the Truffle team implements next!)
```

Comment:
The `Whatever the Truffle team implements next` comment seems too on the nose for something publicly available to me. Change to `Stay tuned` or something similar?